atlp-rwanda/code-ninjas-be
# #181414701 ch(Setup unit testing) Setup unit testing with Mocha & chai

### What does this PR do?
 Setup unit testing
### Description of Task to be completed?
To ensure the back end has all functionality working as intended. You may use Mocha as the test runner and `chai` and `chai-http` for the unit tests.

### How should this be manually tested?
Get started By Running commands bellow:
  - clone repo
  - switch to [ch-setup-test-181414759](https://github.com/atlp-rwanda/code-ninjas-be/tree/ch-setup-test-181414701) branch
  - ``yarn install``: to install all dependencies
  - Create .env file
  - Set up .env as stated in .env example
  - run `yarn migrate` to set up table
  -  run `yarn seed` to populate table with sample data
  - run `yarn test` to start unit testing
  
  - After testing completed successfully you will be able to see passed and failed test, and you will be able to see NYC report .

### Any background context you want to provide?
### What are the relevant pivotal tracker stories?
[#181414701](https://www.pivotaltracker.com/story/show/181414701)
